gpgopt.py

```diff
@@ -0,0 +1,106 @@
+import gnupg
+from pathlib import Path
+from os import getlogin
+from random import randint
+
+
+class GpgOpt():
+    """Initial gnupg, create, import, and export key pairs.
+
+    create, import, and export key pairs.
+    Also, can encrypt and decrypt message.
+    """
+
+    def __init__(self, mygnupghome='gnupgkeys', email=getlogin() + '@scmail.dev'):
+        """
+        """
+        p = Path(mygnupghome)
+        if p.exists() is False:
+            p.mkdir(parents=True)
+
+        self.gpg = gnupg.GPG(gnupghome=mygnupghome)
+        self.email = email
+
+
+    def import_key(self, path):
+        """Check whether the gnupg pub or pvt key already exist.
+        If exist, import it, if not, return false.
+        """
+        key_data = open(path).read()
+
+
+    def create(self, password, key_type="RSA", key_length=1024, expire_date="2y"):
+        """Create a new key pair. need parameters:
+
+        Keyword arguments:
+        password: required. the password of your pvt key.
+        key_type: optional, algorithms to generate the key.
+        key_length: optional, 
+        """
+        # all parameters that user can choose.
+        input_data = self.gpg.gen_key_input(name_email=self.email, key_type=key_type, passphrase=password,
+                                            key_length=key_length, expire_date=expire_date)
+
+        # generate key pair
+        return self.gpg.gen_key(input_data)
+
+
+    def export_key(self, fingerprint, export_pvt=False, password=None, to_file=False, path=None):
+        """Export key pair from the gnupg."""
+        sc_pub = self.gpg.export_keys(fingerprint)
+        sc_pvt = None
+
+        if export_pvt is True and password is not None:
+            sc_pvt = self.gpg.export_keys(fingerprint, True, passphrase=password)
+
+        if to_file is True:
+            try:
+                # store key to file
+                with path.open('w') as f:
+                    f.write(sc_pub)
+                    f.write(sc_pvt)
+            except:
```

Comment:
As a rule of thumb, it's generally considered bad practice to catch _all_ errors while stating that a specific error happened. For instance this would fail if the OS was out of space, or the program doesn't have permission to open the file.
```suggestion
            except FileNotFoundError:
```

Note: Because this immediately raises another exception, it would probably be better to log an error message and quit the program sanely or not catch exceptions at all.

Reply:
Thanks. Because the client also has a logging requirement, I plan to log an error message and quit the program sanely for all possible errors. Is that work for you?